extensions/amp-video/0.1/flexible-bitrate.js

```diff
@@ -241,6 +241,20 @@ export class BitrateManager {
       video.load();
     }
   }
+
+  /**
+   * Called when a video is replaced.
+   * @param {!Element} oldVideo
+   * @param {!Element} newVideo
+   */
+  videoReplaced(oldVideo, newVideo) {
+    if (oldVideo.waitUnlisten_) {
+      oldVideo.waitUnlisten_();
+    }
+    this.videos_ = this.videos_.filter((v) => v.deref() !== oldVideo);
+    this.manage(newVideo);
+    this.sortSources_(newVideo);
```

Comment:
The code that adds sources should call `changedSources` itself, this line would be introducing a different way to use this Class where you don't have to call it manually. Let's keep all code paths/usage similar.